app/modules/inventory/components/inventory_browser.svelte

```diff
@@ -94,26 +101,30 @@
   $: onChange(itemsIds, setupPagination)
 </script>
 
-<InventoryBrowserControls
-  {waitForInventoryData}
-  bind:facetsSelectors
-  bind:facetsSelectedValues
-  bind:textFilterItemsIds
-  {intersectionWorkUris}
-/>
-
-{#await waitForInventoryData}
-  <div class="spinner-wrap">
-    <Spinner center={true} />
-  </div>
-{:then}
-  <PaginatedItems
-    {Component}
-    {componentProps}
-    {pagination}
-    haveSeveralOwners={groupId != null}
+{#if showInventoryWelcome}
+  <InventoryWelcome />
+{:else}
+  <InventoryBrowserControls
+    {waitForInventoryData}
+    bind:facetsSelectors
+    bind:facetsSelectedValues
+    bind:textFilterItemsIds
+    {intersectionWorkUris}
   />
-{/await}
+
+  {#await waitForInventoryData}
+    <div class="spinner-wrap">
+      <Spinner center={true} />
+    </div>
+  {:then}
+    <PaginatedItems
```

Comment:
If i go to friends & group tab and dont wait long enough before going back to my inventory, then 'Nothing to show here' is displayed instead of my items. Then if i scroll down, items are actually displayed. Only happen on large screen.

Reply:
Would it be a good time to make inventory layout component nested inside its parents (user, groups at least, since public is a bit specific), instead of modifying the state of the inventory broswer every time a tab is changed?

Reply:
The good thing with this rewrite is that those main-user/network sections are largely independent (compared to the previous implementation where the same div was reused to display items for the 2 different sections), so we can confidently rule out that it has to do with the transition from one section to the other.

I was able to reproduce the issue with the main user inventory browser alone (which made debugging much less painful ^^), and apparently fix the problem 2200c9a, but I still would have trouble explaining the exact sequence of events that led to that state ><